chain/lf3/ec.go

```diff
@@ -75,57 +80,38 @@ func (ec *ecWrapper) GetTipsetByEpoch(ctx context.Context, epoch int64) (ec.TipS
 	if err != nil {
 		return nil, xerrors.Errorf("getting tipset by height: %w", err)
 	}
-	return wrapTS(ts), nil
+	return &f3TipSet{TipSet: ts}, nil
 }
 
 func (ec *ecWrapper) GetTipset(ctx context.Context, tsk gpbft.TipSetKey) (ec.TipSet, error) {
-	tskLotus, err := types.TipSetKeyFromBytes(tsk)
-	if err != nil {
-		return nil, xerrors.Errorf("decoding tsk: %w", err)
-	}
-
-	ts, err := ec.ChainStore.GetTipSetFromKey(ctx, tskLotus)
+	ts, err := ec.getTipSetFromF3TSK(ctx, tsk)
 	if err != nil {
 		return nil, xerrors.Errorf("getting tipset by key: %w", err)
 	}
 
-	return wrapTS(ts), nil
+	return &f3TipSet{TipSet: ts}, nil
 }
 
-func (ec *ecWrapper) GetHead(_ context.Context) (ec.TipSet, error) {
-	return wrapTS(ec.ChainStore.GetHeaviestTipSet()), nil
+func (ec *ecWrapper) GetHead(context.Context) (ec.TipSet, error) {
+	return &f3TipSet{TipSet: ec.ChainStore.GetHeaviestTipSet()}, nil
```

Comment:
IIRC, there's an issue where `GetHeaviestTipSet` can return nil if we have some form of datastore issue? I'd handle that case here and return an error, just to be safe.